tests/test_unit/test_visualization.py

```diff
@@ -0,0 +1,172 @@
+from unittest.mock import MagicMock, patch
+
+import numpy as np
+import pytest
+import torch
+
+from crabs.detection_tracking.visualization import (
+    draw_bbox,
+    draw_detection,
+    save_images_with_boxes,
+)
+
+
+@pytest.fixture
+def sample_image():
+    # Create a sample image for testing
+    return np.zeros((100, 100, 3), dtype=np.uint8)
+
+
+@pytest.mark.parametrize(
+    "top_left, bottom_right, color, label_text",
+    [
+        ((10, 10), (50, 50), (0, 255, 0), None),
+        ((10, 10), (50, 50), (0, 255, 0), "Test Label"),
+    ],
+)
+def test_draw_bbox(sample_image, top_left, bottom_right, color, label_text):
+    draw_bbox(sample_image, top_left, bottom_right, color)
+
+    if label_text:
+        assert np.any(
+            sample_image[10:20, 10:100] == color
+        )  # Check if label text is drawn
+    else:
+        assert np.any(
+            sample_image[
+                top_left[1] : bottom_right[1], top_left[0] : bottom_right[0]
+            ]
+            == color
+        )  # Check if bounding box is drawn
+
+
+@pytest.mark.parametrize(
+    "top_left, bottom_right, color",
+    [
+        ((10, 10), (15, 15), (0, 255, 0)),
```

Comment:
Since we are only checking one case I don't think this needs parametrisation for now. 

Or we can add another case (but note it will only work for green boxes).

We can expand in a future PR tho, happy to leave as is!